src/frontend/screens/Track/MapPreview.tsx

```diff
@@ -0,0 +1,85 @@
+import React, {FC} from 'react';
+import {StyleSheet} from 'react-native';
+import MapboxGL from '@rnmapbox/maps';
+import {LocationHistoryPoint} from '../../sharedTypes/location.ts';
+import Mapbox from '@rnmapbox/maps';
+import {TrackMapLayer} from '../../sharedComponents/TrackMapLayer.tsx';
+interface TrackScreenMapPreview {
+  locationHistory: LocationHistoryPoint[];
+}
+
+const MAP_STYLE = Mapbox.StyleURL.Outdoors;
+
+export const MapPreview: FC<TrackScreenMapPreview> = ({locationHistory}) => {
+  const [swBoundary, neBoundary] = getAdjustedBounds(locationHistory);
+  return (
+    <MapboxGL.MapView
+      style={styles.map}
+      zoomEnabled={false}
+      logoEnabled={false}
+      scrollEnabled={false}
+      pitchEnabled={false}
+      rotateEnabled={false}
+      compassEnabled={false}
+      scaleBarEnabled={false}
+      styleURL={MAP_STYLE}>
+      <MapboxGL.Camera
+        animationMode="none"
+        zoomLevel={12}
+        bounds={{
+          ne: neBoundary!,
+          sw: swBoundary!,
+          paddingTop: 25,
+          paddingRight: 25,
+          paddingLeft: 25,
+          paddingBottom: 25,
+        }}
+      />
+      <TrackMapLayer locationHistory={locationHistory} />
+    </MapboxGL.MapView>
+  );
+};
+
+const MAP_HEIGHT = 250;
+// Minimum bound size to ensure sufficient map detail
+const MIN_BOUND_SIZE = 0.0003;
+
+const getAdjustedBounds = (locationHistory: LocationHistoryPoint[]) => {
```

Comment:
If we don't use the MIN_BOUND_SIZE on really small observation tracks (aka the ones i tested without leaving my house) it is really zoomed in. Because it is so zoomed in, you cant see any details, you just see a grey background. It makes the map look broken and like no map is rendering.

Reply:
Hmm, im not actually able to make this work. I have tried different combination of the switching the order for lat, but it is not giving me the correct bounding box (and also sometimes breaking the app). Would you spend 10 mins trying bbox on your device?

Reply:
yeah this was my mistake in terms of pointing you to where to look. you need to install a couple of small deps. usage will look like the following:

```ts
import {lineString} from '@turf/helpers';
import {bbox} from '@turf/bbox';

const boundary = bbox(
  lineString(locationHistory.map(point => [point.longitude, point.latitude])),
);
```